test/validation/api/byteorder/byteorder.c

```diff
@@ -0,0 +1,107 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2024 Nokia
+ */
+
+#include <odp_api.h>
+#include <odp_cunit_common.h>
+
+#include <stdint.h>
+
+static void test_defines(void)
+{
+	/* Endianness */
+	CU_ASSERT(ODP_BIG_ENDIAN || ODP_LITTLE_ENDIAN);
+
+	if (ODP_BIG_ENDIAN) {
+		CU_ASSERT(ODP_BYTE_ORDER == ODP_BIG_ENDIAN);
+		CU_ASSERT(!ODP_LITTLE_ENDIAN);
+	}
+
+	if (ODP_LITTLE_ENDIAN) {
+		CU_ASSERT(ODP_BYTE_ORDER == ODP_LITTLE_ENDIAN);
+		CU_ASSERT(!ODP_BIG_ENDIAN);
+	}
+
+	/* Bitfield endianness */
+	CU_ASSERT(ODP_BIG_ENDIAN_BITFIELD || ODP_LITTLE_ENDIAN_BITFIELD);
+
+	if (ODP_BIG_ENDIAN_BITFIELD) {
+		CU_ASSERT(ODP_BITFIELD_ORDER == ODP_BIG_ENDIAN_BITFIELD);
+		CU_ASSERT(!ODP_LITTLE_ENDIAN_BITFIELD);
+	}
+
+	if (ODP_LITTLE_ENDIAN_BITFIELD) {
+		CU_ASSERT(ODP_BITFIELD_ORDER == ODP_LITTLE_ENDIAN_BITFIELD);
+		CU_ASSERT(!ODP_BIG_ENDIAN_BITFIELD);
+	}
+}
+
+static void test_types(void)
+{
+	const uint16_t u16_val = 0xff00;
+	const uint32_t u32_val = 0xffff0000;
+	const uint64_t u64_val = 0xffffffff00000000;
+	const uint16_t u16_val_conv = 0x00ff;
+	const uint32_t u32_val_conv = 0x0000ffff;
+	const uint64_t u64_val_conv = 0x00000000ffffffff;
```

Comment:
For better test coverage the value could be defined as e.g.:
u32_val = 0x12345678;
u32_val_conv = 0x78563412;

Reply:
Changed in V2.